src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -1000,22 +1007,40 @@ protected void handleEvent(JSONObject raw)
     }
 
     @Override
-    public void onBinaryMessage(WebSocket websocket, byte[] binary) throws UnsupportedEncodingException, DataFormatException
+    public void onBinaryMessage(WebSocket websocket, byte[] binary) throws IOException
     {
+        readBuffer.write(binary);
+        //check whether the transition is finished or not
+        if (readBuffer.size() < 4)
+            return;
+        final int suffix = getInt(binary, binary.length - 4);
+        if (suffix != ZLIB_SUFFIX)
+            return;
         //Thanks to ShadowLordAlpha and Shredder121 for code and debugging.
         //Get the compressed message and inflate it
-        ByteArrayOutputStream out = new ByteArrayOutputStream(binary.length * 2);
+        ByteArrayOutputStream out = new ByteArrayOutputStream(readBuffer.size() * 2);
         try (InflaterOutputStream decompressor = new InflaterOutputStream(out, zlibContext))
         {
-            decompressor.write(binary);
+            readBuffer.writeTo(decompressor);
+            // send the inflated message to the TextMessage method
+            onTextMessage(websocket, out.toString("UTF-8"));
         }
         catch (IOException e)
         {
-            throw (DataFormatException) new DataFormatException("Malformed").initCause(e);
+            throw (IOException) new DataFormatException("Malformed").initCause(e);
```

Comment:
```
java.lang.ClassCastException: java.util.zip.DataFormatException cannot be cast to java.io.IOException
	at net.dv8tion.jda.core.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:1032)
	at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:368)
	at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:270)
	at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:990)
	at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:749)
	at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108)
	at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64)
	at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45)
```

Reply:
(line count doesn't match exactly, but this is the culprit)

Reply:
That change was unintended and is already fixed in the next commit I'm working on